### PR TITLE
Implementing rsync

### PR DESCRIPTION
**Pull request by rpadovani**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
+paramiko
```
Comment (alessandrobessi) [inline requirements.txt]: Missing version.

Comment (rpadovani) [reply requirements.txt]: Fixed, thanks

File: main.py [hunk 2]
```diff
@@ -1,17 +1,25 @@
-import boto3
 import argparse
 import logging
 
 from AwsResources import AwsResources
+from Ssh import Ssh
+from rsync import perform_rsync
 
 parser = argparse.ArgumentParser(description='Load data on an AWS EBS volume')
```
Comment (alessandrobessi) [inline main.py:8]: Sometimes the `help` strings begin with capital letters, sometimes they don't.

**Review by alessandrobessi — CHANGES_REQUESTED** [review]
It does not look so broken! Good job!

File: Ssh.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        self.exec_command(f'sudo mkfs -t {filesystem} /dev/sdk')
+
+    def mount_volume(self):
+        logging.info('Mounting EBS')
+        self.exec_command('sudo mkdir /ebs-cargo-data')
+        self.exec_command('sudo mount /dev/sdk /ebs-cargo-data')
+
+    def create_directory(self, dst: str):
+        logging.info('Creating directory')
+        self.exec_command(f'sudo mkdir -p /ebs-cargo-data/{dst}')
```
Comment (lucagrementieri) [inline Ssh.py]: pep8: you should add a newline at the end of file.

File: main.py [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
+parser.add_argument('source', metavar='src', type=str, help='Your source directory')
+parser.add_argument('--dst', help='Specify a directory as destination. If None, src will be saved '
+                                  'in the root directory')
 parser.add_argument('--existing-volume', help='specifies an already existing EBS volume by its '
```
Comment (lucagrementieri) [inline main.py:12]: Isn't it possible to automatically check if the volume already exists by name? Is the volume name a unique identifier?

**Review by lucagrementieri — CHANGES_REQUESTED** [review]

File: main.py (hunk 4 above)
Comment (rpadovani) [reply main.py:12]: No, it's not, you can have multiple EBS with the same name, in the end a name is just an AWS tag, you can also don't have it

File: Ssh.py (hunk 3 above)
Comment (rpadovani) [reply Ssh.py]: Done, thanks, I will also implement CI for this

File: main.py (hunk 2 above)
Comment (rpadovani) [reply main.py:8]: Now the description of the program has a capital letter, all the descriptions do not

**Review by lucagrementieri — APPROVED** [review]

**Review by alessandrobessi — APPROVED** [review]

**Pull request merged**